### PR TITLE
Add mapping function option

### PR DESCRIPTION
**Pull request by mguihal**
Permits to call a function every time a tag is found.

### CODE REVIEW

File: src/json-template.js [hunk 1]
```diff
@@ -4,17 +4,27 @@ var _ = require('lodash');
...
+module.exports = function(templateString, options) {
   var reg = /\"(\{\{([\s\S]+?)\}\})\"/g;
+  options = options || {};
+
   return function(data) {
     return templateString.replace(reg, function(match, tag, key) {
-      var value = _.get(data, _.trim(key));
+      var value = null;
+
+      if (options.mapper) {
```
Comment (FGRibreau) [inline src/json-template.js]: un petit if-exp ? ^^

File: src/json-template.js [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+ * @param  {object} options
+ *  strictMode: if true, throws an error if a key is missing
+ *  mapper: function to execute on each tag
  * @return {function} function(data): string
  */
-module.exports = function(templateString, strictMode) {
+module.exports = function(templateString, options) {
   var reg = /\"(\{\{([\s\S]+?)\}\})\"/g;
+  options = options || {};
+
```
Comment (FGRibreau) [inline src/json-template.js:16]:
```
var mapper = _.isFunction(options.mapper) ? _.partialRight(options.mapper, data, options) : _.flow(_.trim, _.partial(_.get, data));
```

File: src/json-template.js [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
   return function(data) {
     return templateString.replace(reg, function(match, tag, key) {
-      var value = _.get(data, _.trim(key));
 
-      if (_.isUndefined(value) && strictMode) {
+      var value = _.isFunction(options.mapper) ? options.mapper(key, data, options) : _.get(data, _.trim(key));
```
Comment (FGRibreau) [inline src/json-template.js]: var value = mapper(key);

File: src/json-template.js (hunk 2 above)
Comment (mguihal) [inline src/json-template.js:16]: Corrigé en dessous @FGRibreau

**Pull request merged**